wamr-compiler/main.c

```diff
@@ -601,8 +601,8 @@ main(int argc, char *argv[])
             LOG_VERBOSE("Set size level to 1 for Windows AOT file");
             option.size_level = 1;
         }
-#if defined(_WIN32) || defined(_WIN32_) || defined(__APPLE__) \
-    || defined(__MACH__)
+#if defined(_WIN32) || defined(_WIN32_) \
+    || ((defined(__APPLE__) || defined(__MACH__)) && !defined(__arm64__))
```

Comment:
This excludes setting size level to 1 when the host os is macos and the host cpu is arm64, is it necessary? In macos aarch64, wamrc generates aot file with size level to 1 by default since it may generate relocations which require the target address to be in range 0-4GB and make them not easy to be applied (as aot loader os_mmap may return address larger than 4GB).
Maybe it is better not to change the code here?